pluggy.py

```diff
@@ -749,10 +761,28 @@ def _maybe_apply_history(self, method):
                     proc(res[0])
 
 
+class HookSpec(object):
```

Comment:
Even if we decide to abandon this defaults idea I **really** think we should keep this encapsulation of `hookspec` objects with a type much like we do with `HookImpl`. It makes introspection tasks down the road that much simpler.